pulsar-broker-common/src/test/java/org/apache/pulsar/jetty/tls/JettySslContextFactoryTest.java

```diff
@@ -0,0 +1,174 @@
+        List<ServerConnector> connectors = new ArrayList<>();
+        SslContextFactory factory = JettySslContextFactory.createServerSslContext(
+                null,
+                false,
+                Resources.getResource("ssl/my-ca/ca.pem").getPath(),
+                Resources.getResource("ssl/my-ca/server-ca.pem").getPath(),
+                Resources.getResource("ssl/my-ca/server-key.pem").getPath(),
+                true,
+                null,
+                null,
+                600);
+
+        ServerConnector connector = new ServerConnector(server, factory);
+        connector.setPort(0);
+        connectors.add(connector);
+        server.setConnectors(connectors.toArray(new ServerConnector[0]));
+        server.start();
+        // client connect
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+        registryBuilder.register("https",
+                new SSLConnectionSocketFactory(getClientSslContext(), new NoopHostnameVerifier()));
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(registryBuilder.build());
+        httpClientBuilder.setConnectionManager(cm);
+        CloseableHttpClient httpClient = httpClientBuilder.build();
+        HttpGet httpGet = new HttpGet("https://localhost:" + connector.getLocalPort());
+        httpClient.execute(httpGet);
+        httpClient.close();
+        server.stop();
+    }
+
+    @Test(expectedExceptions = SSLHandshakeException.class)
+    public void testJettyTlsServerInvalidTlsProtocol() throws Exception {
+        Server server = new Server();
+        List<ServerConnector> connectors = new ArrayList<>();
+        SslContextFactory factory = JettySslContextFactory.createServerSslContext(
+                null,
+                false,
+                Resources.getResource("ssl/my-ca/ca.pem").getPath(),
+                Resources.getResource("ssl/my-ca/server-ca.pem").getPath(),
+                Resources.getResource("ssl/my-ca/server-key.pem").getPath(),
+                true,
+                null,
+                new HashSet<String>() {
+                    {
+                        this.add("TLSv1.3");
+                    }
+                },
+                600);
+        factory.setHostnameVerifier((s, sslSession) -> true);
+        ServerConnector connector = new ServerConnector(server, factory);
+        connector.setPort(0);
+        connectors.add(connector);
+        server.setConnectors(connectors.toArray(new ServerConnector[0]));
+        server.start();
+        // client connect
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
+        registryBuilder.register("https", new SSLConnectionSocketFactory(getClientSslContext(),
+                new String[]{"TLSv1.2"}, null, new NoopHostnameVerifier()));
```

Comment:
I tried to change `TLSv1.2` to `TLSv1.3`, the test still passed. Is it expected?

Reply:
It should be failed. Do you use JAVA 11?

Reply:
This is my test that change `TLSv1.2` to `TLSv1.3`
```
/Library/Java/JavaVirtualMachines/openjdk-11.jdk/Contents/Home/bin/java -ea -DtestRealAWS=false -DtestRetryCount=1 -Didea.test.cyclic.buffer.size=1048576 -javaagent:/Users/nodece/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=51037:/Users/nodece/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/nodece/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app/Contents/lib/idea_rt.jar:/Users/nodece/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app/Contents/plugins/testng/lib/testng-rt.jar:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-broker-common/target/test-classes:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-broker-common/target/classes:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-zookeeper-utils/target/classes:/Users/nodece/.m2/repository/org/apache/commons/commons-lang3/3.11/commons-lang3-3.11.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-server/4.14.4/bookkeeper-server-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-server/4.14.4/stream-storage-server-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-service-impl/4.14.4/stream-storage-service-impl-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-service-api/4.14.4/stream-storage-service-api-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-proto/4.14.4/stream-storage-proto-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-java-client-base/4.14.4/stream-storage-java-client-base-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-api/4.14.4/stream-storage-api-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/statelib/4.14.4/statelib-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/distributedlog/distributedlog-core/4.14.4/distributedlog-core-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/distributedlog/distributedlog-protocol/4.14.4/distributedlog-protocol-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/distributedlog/distributedlog-common/4.14.4/distributedlog-common-4.14.4.jar:/Users/nodece/.m2/repository/net/jpountz/lz4/lz4/1.3.0/lz4-1.3.0.jar:/Users/nodece/.m2/repository/org/apache/thrift/libthrift/0.14.2/libthrift-0.14.2.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-common/4.14.4/stream-storage-common-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/distributedlog/distributedlog-core/4.14.4/distributedlog-core-4.14.4-tests.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stream-storage-java-client/4.14.4/stream-storage-java-client-4.14.4.jar:/Users/nodece/.m2/repository/com/google/auth/google-auth-library-credentials/0.22.2/google-auth-library-credentials-0.22.2.jar:/Users/nodece/.m2/repository/com/google/android/annotations/4.1.1.4/annotations-4.1.1.4.jar:/Users/nodece/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.19/animal-sniffer-annotations-1.19.jar:/Users/nodece/.m2/repository/com/google/protobuf/protobuf-java-util/3.19.2/protobuf-java-util-3.19.2.jar:/Users/nodece/.m2/repository/io/netty/netty-codec-http2/4.1.76.Final/netty-codec-http2-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-codec-http/4.1.76.Final/netty-codec-http-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-handler-proxy/4.1.76.Final/netty-handler-proxy-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-codec-socks/4.1.76.Final/netty-codec-socks-4.1.76.Final.jar:/Users/nodece/.m2/repository/com/google/api/grpc/proto-google-common-protos/2.0.1/proto-google-common-protos-2.0.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-protobuf-lite/1.45.1/grpc-protobuf-lite-1.45.1.jar:/Users/nodece/.m2/repository/com/google/auto/value/auto-value-annotations/1.7.4/auto-value-annotations-1.7.4.jar:/Users/nodece/.m2/repository/io/opencensus/opencensus-api/0.28.0/opencensus-api-0.28.0.jar:/Users/nodece/.m2/repository/io/grpc/grpc-alts/1.45.1/grpc-alts-1.45.1.jar:/Users/nodece/.m2/repository/org/conscrypt/conscrypt-openjdk-uber/2.5.2/conscrypt-openjdk-uber-2.5.2.jar:/Users/nodece/.m2/repository/com/google/auth/google-auth-library-oauth2-http/0.22.2/google-auth-library-oauth2-http-0.22.2.jar:/Users/nodece/.m2/repository/com/google/http-client/google-http-client/1.41.0/google-http-client-1.41.0.jar:/Users/nodece/.m2/repository/io/opencensus/opencensus-contrib-http-util/0.28.0/opencensus-contrib-http-util-0.28.0.jar:/Users/nodece/.m2/repository/com/google/http-client/google-http-client-jackson2/1.41.0/google-http-client-jackson2-1.41.0.jar:/Users/nodece/.m2/repository/com/google/re2j/re2j/1.5/re2j-1.5.jar:/Users/nodece/.m2/repository/io/opencensus/opencensus-proto/0.2.0/opencensus-proto-0.2.0.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/http/vertx-http-server/4.14.4/vertx-http-server-4.14.4.jar:/Users/nodece/.m2/repository/io/vertx/vertx-core/3.9.8/vertx-core-3.9.8.jar:/Users/nodece/.m2/repository/io/vertx/vertx-web/3.9.8/vertx-web-3.9.8.jar:/Users/nodece/.m2/repository/io/vertx/vertx-web-common/3.9.8/vertx-web-common-3.9.8.jar:/Users/nodece/.m2/repository/io/vertx/vertx-auth-common/3.9.8/vertx-auth-common-3.9.8.jar:/Users/nodece/.m2/repository/io/vertx/vertx-bridge-common/3.9.8/vertx-bridge-common-3.9.8.jar:/Users/nodece/.m2/repository/org/xerial/snappy/snappy-java/1.1.7/snappy-java-1.1.7.jar:/Users/nodece/.m2/repository/io/dropwizard/metrics/metrics-core/3.2.5/metrics-core-3.2.5.jar:/Users/nodece/.m2/repository/org/apache/curator/curator-recipes/5.1.0/curator-recipes-5.1.0.jar:/Users/nodece/.m2/repository/org/apache/curator/curator-framework/5.1.0/curator-framework-5.1.0.jar:/Users/nodece/.m2/repository/org/apache/curator/curator-client/5.1.0/curator-client-5.1.0.jar:/Users/nodece/.m2/repository/io/grpc/grpc-all/1.45.1/grpc-all-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-api/1.45.1/grpc-api-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-auth/1.45.1/grpc-auth-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-context/1.45.1/grpc-context-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-core/1.45.1/grpc-core-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-grpclb/1.45.1/grpc-grpclb-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-netty/1.45.1/grpc-netty-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-protobuf/1.45.1/grpc-protobuf-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-rls/1.45.1/grpc-rls-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-services/1.45.1/grpc-services-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-stub/1.45.1/grpc-stub-1.45.1.jar:/Users/nodece/.m2/repository/io/grpc/grpc-xds/1.45.1/grpc-xds-1.45.1.jar:/Users/nodece/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.67/bcpkix-jdk15on-1.67.jar:/Users/nodece/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.67/bcprov-jdk15on-1.67.jar:/Users/nodece/.m2/repository/io/grpc/grpc-netty-shaded/1.45.1/grpc-netty-shaded-1.45.1.jar:/Users/nodece/.m2/repository/io/perfmark/perfmark-api/0.19.0/perfmark-api-0.19.0.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stats/prometheus-metrics-provider/4.14.4/prometheus-metrics-provider-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/stats/bookkeeper-stats-api/4.14.4/bookkeeper-stats-api-4.14.4.jar:/Users/nodece/.m2/repository/io/prometheus/simpleclient/0.5.0/simpleclient-0.5.0.jar:/Users/nodece/.m2/repository/io/prometheus/simpleclient_hotspot/0.5.0/simpleclient_hotspot-0.5.0.jar:/Users/nodece/.m2/repository/io/prometheus/simpleclient_servlet/0.5.0/simpleclient_servlet-0.5.0.jar:/Users/nodece/.m2/repository/io/prometheus/simpleclient_common/0.5.0/simpleclient_common-0.5.0.jar:/Users/nodece/.m2/repository/io/netty/netty-common/4.1.76.Final/netty-common-4.1.76.Final.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-servlet/9.4.44.v20210927/jetty-servlet-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-security/9.4.44.v20210927/jetty-security-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-server/9.4.44.v20210927/jetty-server-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-http/9.4.44.v20210927/jetty-http-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-io/9.4.44.v20210927/jetty-io-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-util-ajax/9.4.44.v20210927/jetty-util-ajax-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/org/eclipse/jetty/jetty-util/9.4.44.v20210927/jetty-util-9.4.44.v20210927.jar:/Users/nodece/.m2/repository/com/yahoo/datasketches/sketches-core/0.8.3/sketches-core-0.8.3.jar:/Users/nodece/.m2/repository/com/yahoo/datasketches/memory/0.8.3/memory-0.8.3.jar:/Users/nodece/.m2/repository/com/github/ben-manes/caffeine/caffeine/2.9.1/caffeine-2.9.1.jar:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-common/target/classes:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-client-api/target/classes:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-client-admin-api/target/classes:/Users/nodece/.m2/repository/io/swagger/swagger-annotations/1.6.2/swagger-annotations-1.6.2.jar:/Users/nodece/.m2/repository/io/netty/netty-resolver-dns/4.1.76.Final/netty-resolver-dns-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-codec-dns/4.1.76.Final/netty-codec-dns-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-transport-native-unix-common/4.1.76.Final/netty-transport-native-unix-common-4.1.76.Final-linux-x86_64.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/cpu-affinity/4.14.4/cpu-affinity-4.14.4.jar:/Users/nodece/.m2/repository/io/airlift/aircompressor/0.20/aircompressor-0.20.jar:/Users/nodece/.m2/repository/io/netty/netty-codec-haproxy/4.1.76.Final/netty-codec-haproxy-4.1.76.Final.jar:/Users/nodece/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.13.2/jackson-dataformat-yaml-2.13.2.jar:/Users/nodece/.m2/repository/com/google/code/gson/gson/2.8.9/gson-2.8.9.jar:/Users/nodece/.m2/repository/io/prometheus/simpleclient_caffeine/0.5.0/simpleclient_caffeine-0.5.0.jar:/Users/nodece/Documents/repo/streamnative/pulsar/pulsar-metadata/target/classes:/Users/nodece/.m2/repository/org/apache/zookeeper/zookeeper/3.8.0/zookeeper-3.8.0.jar:/Users/nodece/.m2/repository/org/apache/zookeeper/zookeeper-jute/3.8.0/zookeeper-jute-3.8.0.jar:/Users/nodece/.m2/repository/org/apache/yetus/audience-annotations/0.12.0/audience-annotations-0.12.0.jar:/Users/nodece/.m2/repository/io/netty/netty-transport-native-epoll/4.1.76.Final/netty-transport-native-epoll-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/etcd/jetcd-core/0.5.11/jetcd-core-0.5.11.jar:/Users/nodece/.m2/repository/io/etcd/jetcd-common/0.5.11/jetcd-common-0.5.11.jar:/Users/nodece/.m2/repository/net/jodah/failsafe/2.4.4/failsafe-2.4.4.jar:/Users/nodece/.m2/repository/com/google/auto/service/auto-service-annotations/1.0/auto-service-annotations-1.0.jar:/Users/nodece/.m2/repository/com/google/guava/guava/31.0.1-jre/guava-31.0.1-jre.jar:/Users/nodece/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/Users/nodece/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/nodece/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/Users/nodece/.m2/repository/org/checkerframework/checker-qual/3.12.0/checker-qual-3.12.0.jar:/Users/nodece/.m2/repository/com/google/errorprone/error_prone_annotations/2.5.1/error_prone_annotations-2.5.1.jar:/Users/nodece/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar:/Users/nodece/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar:/Users/nodece/.m2/repository/javax/ws/rs/javax.ws.rs-api/2.1/javax.ws.rs-api-2.1.jar:/Users/nodece/.m2/repository/io/jsonwebtoken/jjwt-impl/0.11.1/jjwt-impl-0.11.1.jar:/Users/nodece/.m2/repository/io/jsonwebtoken/jjwt-api/0.11.1/jjwt-api-0.11.1.jar:/Users/nodece/.m2/repository/io/jsonwebtoken/jjwt-jackson/0.11.1/jjwt-jackson-0.11.1.jar:/Users/nodece/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.13.2.1/jackson-databind-2.13.2.1.jar:/Users/nodece/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.2/jackson-annotations-2.13.2.jar:/Users/nodece/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.13.2/jackson-core-2.13.2.jar:/Users/nodece/.m2/repository/org/awaitility/awaitility/4.2.0/awaitility-4.2.0.jar:/Users/nodece/.m2/repository/org/hamcrest/hamcrest/2.2/hamcrest-2.2.jar:/Users/nodece/Documents/repo/streamnative/pulsar/buildtools/target/classes:/Users/nodece/.m2/repository/org/yaml/snakeyaml/1.30/snakeyaml-1.30.jar:/Users/nodece/.m2/repository/org/apache/ant/ant/1.10.12/ant-1.10.12.jar:/Users/nodece/.m2/repository/org/apache/ant/ant-launcher/1.10.12/ant-launcher-1.10.12.jar:/Users/nodece/.m2/repository/com/google/inject/guice/5.1.0/guice-5.1.0.jar:/Users/nodece/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar:/Users/nodece/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar:/Users/nodece/.m2/repository/junit/junit/4.13.1/junit-4.13.1.jar:/Users/nodece/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/Users/nodece/.m2/repository/org/apache/logging/log4j/log4j-api/2.17.1/log4j-api-2.17.1.jar:/Users/nodece/.m2/repository/org/apache/logging/log4j/log4j-core/2.17.1/log4j-core-2.17.1.jar:/Users/nodece/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.17.1/log4j-slf4j-impl-2.17.1.jar:/Users/nodece/.m2/repository/org/apache/logging/log4j/log4j-1.2-api/2.17.1/log4j-1.2-api-2.17.1.jar:/Users/nodece/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.32/jcl-over-slf4j-1.7.32.jar:/Users/nodece/.m2/repository/org/testng/testng/7.3.0/testng-7.3.0.jar:/Users/nodece/.m2/repository/com/beust/jcommander/1.78/jcommander-1.78.jar:/Users/nodece/.m2/repository/com/google/inject/guice/4.2.2/guice-4.2.2-no_aop.jar:/Users/nodece/.m2/repository/org/mockito/mockito-core/3.12.4/mockito-core-3.12.4.jar:/Users/nodece/.m2/repository/net/bytebuddy/byte-buddy/1.11.13/byte-buddy-1.11.13.jar:/Users/nodece/.m2/repository/net/bytebuddy/byte-buddy-agent/1.11.13/byte-buddy-agent-1.11.13.jar:/Users/nodece/.m2/repository/org/objenesis/objenesis/3.1/objenesis-3.1.jar:/Users/nodece/.m2/repository/org/mockito/mockito-inline/3.12.4/mockito-inline-3.12.4.jar:/Users/nodece/.m2/repository/org/powermock/powermock-reflect/2.0.9/powermock-reflect-2.0.9.jar:/Users/nodece/.m2/repository/org/assertj/assertj-core/3.18.1/assertj-core-3.18.1.jar:/Users/nodece/.m2/repository/org/projectlombok/lombok/1.18.22/lombok-1.18.22.jar:/Users/nodece/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-server/4.14.4/bookkeeper-server-4.14.4-tests.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-common/4.14.4/bookkeeper-common-4.14.4.jar:/Users/nodece/.m2/repository/org/jctools/jctools-core/2.1.2/jctools-core-2.1.2.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-common-allocator/4.14.4/bookkeeper-common-allocator-4.14.4.jar:/Users/nodece/.m2/repository/io/netty/netty-buffer/4.1.76.Final/netty-buffer-4.1.76.Final.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-proto/4.14.4/bookkeeper-proto-4.14.4.jar:/Users/nodece/.m2/repository/com/google/protobuf/protobuf-java/3.19.2/protobuf-java-3.19.2.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/bookkeeper-tools-framework/4.14.4/bookkeeper-tools-framework-4.14.4.jar:/Users/nodece/.m2/repository/org/rocksdb/rocksdbjni/6.10.2/rocksdbjni-6.10.2.jar:/Users/nodece/.m2/repository/io/netty/netty-handler/4.1.76.Final/netty-handler-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-resolver/4.1.76.Final/netty-resolver-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-transport/4.1.76.Final/netty-transport-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-codec/4.1.76.Final/netty-codec-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-transport-native-epoll/4.1.76.Final/netty-transport-native-epoll-4.1.76.Final-linux-x86_64.jar:/Users/nodece/.m2/repository/io/netty/netty-transport-native-unix-common/4.1.76.Final/netty-transport-native-unix-common-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-transport-classes-epoll/4.1.76.Final/netty-transport-classes-epoll-4.1.76.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-classes/2.0.51.Final/netty-tcnative-classes-2.0.51.Final.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-linux-x86_64.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-linux-aarch_64.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-osx-x86_64.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-osx-aarch_64.jar:/Users/nodece/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.51.Final/netty-tcnative-boringssl-static-2.0.51.Final-windows-x86_64.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/http/http-server/4.14.4/http-server-4.14.4.jar:/Users/nodece/.m2/repository/org/apache/bookkeeper/circe-checksum/4.14.4/circe-checksum-4.14.4.jar:/Users/nodece/.m2/repository/commons-cli/commons-cli/1.5.0/commons-cli-1.5.0.jar:/Users/nodece/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar:/Users/nodece/.m2/repository/commons-io/commons-io/2.8.0/commons-io-2.8.0.jar:/Users/nodece/.m2/repository/org/apache/commons/commons-collections4/4.1/commons-collections4-4.1.jar:/Users/nodece/.m2/repository/net/java/dev/jna/jna/4.2.0/jna-4.2.0.jar:/Users/nodece/.m2/repository/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar:/Users/nodece/.m2/repository/org/apache/httpcomponents/httpcore/4.4.15/httpcore-4.4.15.jar:/Users/nodece/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/Users/nodece/.m2/repository/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar:/Users/nodece/.m2/repository/commons-configuration/commons-configuration/1.10/commons-configuration-1.10.jar:/Users/nodece/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar com.intellij.rt.testng.RemoteTestNGStarter -usedefaultlisteners false -socket51036 @w@/private/var/folders/y4/vy8n0vg9427_rh7976wd675h0000gn/T/idea_working_dirs_testng.tmp -temp /private/var/folders/y4/vy8n0vg9427_rh7976wd675h0000gn/T/idea_testng.tmp
2022-04-27T14:53:10,909+0800 INFO  [main] o.e.j.u.log@170 - Logging initialized @1346ms to org.eclipse.jetty.util.log.Slf4jLog {}
2022-04-27T14:53:11,514+0800 INFO  [main] o.e.j.s.Server@375 - jetty-9.4.44.v20210927; built: 2021-09-27T23:02:44.612Z; git: 8da83308eeca865e495e53ef315a249d63ba9332; jvm 11.0.12+0 {}
2022-04-27T14:53:11,865+0800 INFO  [main] o.e.j.s.AbstractConnector@331 - Started ServerConnector@7dc51783{SSL, (ssl, http/1.1)}{0.0.0.0:51041} {}
2022-04-27T14:53:11,865+0800 INFO  [main] o.e.j.s.Server@415 - Started @2307ms {}
2022-04-27T14:53:12,283+0800 INFO  [main] o.e.j.s.AbstractConnector@381 - Stopped ServerConnector@7dc51783{SSL, (ssl, http/1.1)}{0.0.0.0:0} {}

org.testng.TestException: 
Method JettySslContextFactoryWithKeyStoreTest.testJettyTlsServerInvalidTlsProtocol()[pri:0, instance:org.apache.pulsar.jetty.tls.JettySslContextFactoryWithKeyStoreTest@58c34bb3] should have thrown an exception of type class javax.net.ssl.SSLHandshakeException

	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:82)
	at org.testng.internal.TestInvoker.considerExceptions(TestInvoker.java:747)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:635)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.testng.TestRunner.privateRun(TestRunner.java:764)
	at org.testng.TestRunner.run(TestRunner.java:585)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)


===============================================
Default Suite
Total tests run: 1, Passes: 0, Failures: 1, Skips: 0
===============================================


Process finished with exit code 0
```

Reply:
It's still a JDK issue. When I switched from 8 to 11, it failed as expected.

Reply:
The KeyStore needs to do a fix when using JAVA 8.

I'll do this next PR.